modules/environments/service_principal.tf

```diff
@@ -0,0 +1,42 @@
+resource "time_rotating" "one_year" {
+  rotation_days = 365
+}
+
+resource "random_uuid" "app_uuid" {}
+
+resource "azuread_application" "app" {
+  display_name = local.app_name
+  api {
+    oauth2_permission_scope {
+      admin_consent_description  = "Allow the application to access ${local.app_name} on behalf of the signed-in user."
+      admin_consent_display_name = join(" ", ["Access", local.app_name])
```

Comment:
can you search join throughout I suspect all of them can be replaced and will be far easier to read